apps/admin-gui/src/app/shared/components/dialogs/add-edit-notification-dialog/add-edit-notification-dialog.component.ts

```diff
@@ -177,31 +177,14 @@ export class AddEditNotificationDialogComponent implements OnInit {
       ? (textarea.children.item(0) as HTMLTextAreaElement)
       : (input.children.item(0) as HTMLInputElement);
     const position: number = place.selectionStart;
-    if (this.isTextFocused) {
-      if (format === 'html') {
-        this.applicationMail.htmlMessage[language].text =
-          this.applicationMail.htmlMessage[language].text.substring(0, position) +
-          tag +
-          this.applicationMail.htmlMessage[language].text.substring(position);
-      } else {
-        this.applicationMail.message[language].text =
-          this.applicationMail.message[language].text.substring(0, position) +
-          tag +
-          this.applicationMail.message[language].text.substring(position);
-      }
-    } else {
-      if (format === 'html') {
-        this.applicationMail.htmlMessage[language].subject =
-          this.applicationMail.htmlMessage[language].subject.substring(0, position) +
-          tag +
-          this.applicationMail.htmlMessage[language].subject.substring(position);
-      } else {
-        this.applicationMail.message[language].subject =
-          this.applicationMail.message[language].subject.substring(0, position) +
-          tag +
-          this.applicationMail.message[language].subject.substring(position);
-      }
+    if (format === 'plain_text') {
+      format = 'plain';
     }
+    const form = this.inputFormGroup.get(
+      `${language}-${format}-${this.isTextFocused ? 'text' : 'subject'}`
+    );
+    const curValue = String(form.value);
```

Comment:
Tags are now added properly, well done. I would just like to see some technical changes.

The formats are defined inside this class and serve no other purpose it seem. So we could redefine the "plain_text" option to "plain" so the conversion is not needed.

Also please set the type of `inputFormGroup` to `FormGroup<Record<string, FormControl<string>>>` (form group of objects that have key of type string and value of type `FormControl<string>`) so we have type checking. This will require same type change to the variable `formGroupFields` in `ngOnInit`. And the conversion of value to string will not be needed anymore.

Reply:
Adjusted the typing, changed `plain_text` to `plain` analogously also in the edit-notification component + removed deprecated syntax in the add-edit-notification component